From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NeonOverflow <nneonoverflow@protonmail.com>
Date: Thu, 13 Jan 2022 22:58:50 -0500
Subject: [PATCH] Added and implemented PlayerLootDropEvent.

As a result of how the PlayerDeathEvent was implemented, when a player dies their items are simply dropped
through the "world.dropItem()" method. This makes it very annoying for plugin developers to ultimately
track down the Item entities dropped when a player dies. This patch implements a "PlayerLootDropEvent",
making it easy for developers to track down the Item entities dropped when a player dies. Furthermore,
Item entities dropped on player death are marked as being thrown by the player that died.

diff --git a/src/main/java/com/destroystokyo/paper/event/player/PlayerLootDropEvent.java b/src/main/java/com/destroystokyo/paper/event/player/PlayerLootDropEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..f8fc418d73f465ee79552f414496615fb223c144
--- /dev/null
+++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerLootDropEvent.java
@@ -0,0 +1,47 @@
+package com.destroystokyo.paper.event.player;
+
+import java.util.List;
+import org.bukkit.entity.Item;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.jetbrains.annotations.NotNull;
+
+public class PlayerLootDropEvent extends PlayerEvent implements Cancellable {
+    private static final HandlerList handlers = new HandlerList();
+    private final List<Item> items;
+    private boolean cancel = false;
+
+    public PlayerLootDropEvent(@NotNull Player player, @NotNull List<Item> items) {
+        super(player);
+        this.items = items;
+    }
+
+    /**
+     * @return the item entities of the loot dropped.
+     */
+    public List<Item> getItems() {
+        return items;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        return cancel;
+    }
+
+    @Override
+    public void setCancelled(boolean cancel) {
+        this.cancel = cancel;
+    }
+
+    @Override
+    @NotNull
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
